SCRD-3501 Added delete compute page as part of replacing a compute server

guglo opened the pull request:
This commit includes:
1. Added delete compute page.
2. Refactored the replace detail page to use ConfirmModal to have
  consistent look and feel as the rest of the dialogs.
3. Renamed  osInstallPassword to be osPassword and make it  available
  without osInstall for replacing a compute.
4. Updated to handle empty IPMI values when select and deselect
installOS for compute server.
5. Always show osUsername and osPassword inputs for compute server
6. Updated to have installOS page when user chooses to installOS
7. Moved some methods related to logging progress to misc utils
8. Fixed inconsistent colon usage in the compelete page

guglo commented:
The change still has to go with https://gerrit.suse.provo.cloud//#/c/5300/

jeremy-moffitt commented:
I'm a little confused by `Added delete compute page.` ... Is this as part of the replace workflow? Or is it supposed to be a standalone action? I don't see a "delete" action anywhere on the computes list

jeremy-moffitt commented:
@guglo thanks for clarifying, based on the original commit message I thought there might be a whole new page for deleting servers (via the menu in the server summary page or similar), I understand now that this is specific to the replace workflow.

jeremy-moffitt commented:
tried it out with the mock backend, works great there, peeking through the code now

jeremy-moffitt reviewed (changes requested):
mostly looks good... couple of questions, including one that I'm pretty sure is a typo that needs to be fixed

guglo commented on src/pages/ReplaceServer/DeleteCompute.js:
```diff
@@ -0,0 +1,564 @@
+} from '../../utils/constants.js';
+import { removeServerFromModel, getMergedServer, genUID } from '../../utils/ModelUtils.js';
+
+const MANUAL_SHUTDOWN = 'manual_shutdonw';
```
oh...will fix

guglo commented on src/pages/ReplaceServer/DeleteCompute.js:
```diff
@@ -0,0 +1,564 @@
+    if (server) {
+      server = server.toJS();
+      let model =
+        removeServerFromModel(server, this.props.model);
```
not sure why...one line should be fine. will update

guglo commented on src/pages/ReplaceServer/DeleteCompute.js:
```diff
@@ -0,0 +1,564 @@
+          if(!server['uid']) {
+            server['uid'] = genUID('manual');
+          }
+          if(!server['manual']) {
```
should be !server['source']...will fix.

jeremy-moffitt reviewed (approved):
LGTM, worked in the dev environment

Commit:
Rebased with latest master

SCRD-3501 Added delete compute page

This commit includes:
1. Added delete compute page.
2. Refactored the replace detail page to use ConfirmModal to have
   consistent look and feel as the rest of the dialogs.
3. Renamed  osInstallPassword to be osPassword and make it  available
   without osInstall for replacing a compute.
4. Updated to handle empty IPMI values when select and deselect
   installOS for compute server.
5. Always show osUsername and osPassword inputs for compute server
6. Updated to have installOS page when user chooses to installOS
7. Moved some methods related to logging progress to misc utils
8. Fixed inconsistent colon usage in the compelete page

GarySmith reviewed (approved):
Looks good, but I noticed a few cleanup items are necessary. I filed SCRD-6374, SCRD-6375, and SCRD-6376 to cover those.
